### PR TITLE
Json File Generation and Rule Failure Extensions (next iteration step)

### PR DESCRIPTION
**Pull request by odanivan**

### CODE REVIEW

**Review by mxrcx — APPROVED** [review]
nice

File: src/components/ActionItem.vue [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
+      v-if="showActionChooser"
       v-on:action="showActionChooser = false; action.name=$event"
       v-on:abort="showActionChooser = false"
     ></ActionChooser>Aktion:
     <div class="action-content">
-      <input
+      <button
         class="action-input"
-        placeholder="Bezeichner eingeben..."
+        placeholder="Bezeichner auswählen..."
```
Comment (hkohlsaat) [inline src/components/ActionItem.vue]: The placeholder attribute does not apply to button element. See [w3schools - placeholder attribute](https://www.w3schools.com/tags/att_placeholder.asp).

File: src/components/ActionItem.vue [hunk 2]
```diff
@@ -1,20 +1,19 @@
...
         v-bind:value="actionName"
         v-on:click="showActionChooser = true"
         list="actions"
-        type="text"
```
Comment (hkohlsaat) [inline src/components/ActionItem.vue:15]: Why don't you keep `type="button"`. See [w3schools - button tag - Definition and Usage](https://www.w3schools.com/tags/tag_button.asp)

File: src/components/ActionItem.vue [hunk 3]
```diff
@@ -1,20 +1,19 @@
...
       v-on:action="showActionChooser = false; action.name=$event"
       v-on:abort="showActionChooser = false"
     ></ActionChooser>Aktion:
     <div class="action-content">
-      <input
+      <button
         class="action-input"
-        placeholder="Bezeichner eingeben..."
+        placeholder="Bezeichner auswählen..."
         v-bind:value="actionName"
```
Comment (hkohlsaat) [inline src/components/ActionItem.vue:12]: Don't need this any more, do we?

File: src/components/ActionItem.vue [hunk 4]
```diff
@@ -57,7 +56,8 @@ export default {
   },
   data: function() {
     return {
-      showActionChooser: true
+      showActionChooser: false,
+      shownOnCreation: false
```
Comment (hkohlsaat) [inline src/components/ActionItem.vue]: try showing after creation:
```
...,
created: function() {
    showActionChooser(); // maybe introduce this function
}, ...
```

File: src/components/BaseButton.vue [hunk 5]
```diff
@@ -8,18 +14,27 @@ export default {
   props: {
     big: Boolean,
     textlike: Boolean,
-    onClick: { type: Function, required: true }
+    onClick: { type: Function, required: true },
+    onClickParam: { default: null, required: false }
```
Comment (hkohlsaat) [inline src/components/BaseButton.vue]: hacky :O always pass event with every function call

File: src/components/RuleEditor.vue [hunk 6]
```diff
@@ -1,7 +1,7 @@
 <template>
   <div class="rule-editor">
     <div>
-      <input class="under-cover" placeholder="Name der Lizenz">
+      <input id="licence-name" class="under-cover" placeholder="Name der Lizenz">
```
Comment (hkohlsaat) [inline src/components/RuleEditor.vue]: make it a class

File: src/components/RuleEditor.vue [hunk 7]
```diff
@@ -20,8 +20,10 @@
     <BaseButton v-bind:onClick="newDuty">Verpflichtung hinzufügen</BaseButton>
     <BaseButton v-bind:onClick="newProhibition">Verbot hinzufügen</BaseButton>
     <hr>
-    <BaseButton big v-bind:onClick="generateLicence">Generate Licence</BaseButton>
-    <div style="min-height: 150px; border: dotted black 1px"><i>divider</i></div>
+    <BaseButton style="margin-bottom: 50px;" big v-bind:onClick="generateLicence">Generate Licence</BaseButton>
```
Comment (hkohlsaat) [inline src/components/RuleEditor.vue:23]: inline styles :O

**Pull request merged**